tools/generate.py

```diff
@@ -339,9 +339,9 @@ def _build(parent: ObjectT, namespace: str, overrides: dict[str, str]) -> str:
     typings = "from typing import Any, Callable, Literal, Optional, Tuple, Type, TypeVar, Sequence"
 
     typevars: list[str] = []
-    imports: list[str] = []
+    imports: list[str] = ["from typing_extensions import deprecated"]
```

Comment:
Maybe use

```suggestion
    imports: list[str] = ["""
try:
    from warnings import deprecated
except ImportError:
    from typing_extensions import deprecated
"""]
```

and 

```patch
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -14,6 +14,10 @@ authors = [
   {email = "reiter.christoph@gmail.com"},
   {name = "Christoph Reiter"}
 ]
+dependencies = [
+    # for @deprecated decorator
+    'typing_extensions>="4.5.0"; python_version<"3.13"',
+]
 classifiers = [
     "Programming Language :: Python :: 3",
     "Intended Audience :: Developers",
```

Reply:
Added the dependency. 

I am not sure about the `warning` import. Is there a difference between `warnings`and `typing_extensions`, beside the minimum python version required?

@lovetox what do you think?

Reply:
not sure what you mean, there is no difference logic/code wise. It enables people that use python 3.13+ to install one less dependency when using pygobject. So looks fine to me to do it like that.